src/components/AnnotationViewer.vue

```diff
@@ -830,6 +810,73 @@ export default class AnnotationViewer extends Vue {
     return [];
   }
 
+  addAnnotation(newAnnotation: IAnnotation) {
+    // Save the new annotation
+    this.annotationStore.addAnnotation(newAnnotation);
+
+    this.addAnnotationConnections(newAnnotation).then(
+      (connections: IAnnotationConnection[]) => {
+        this.propertiesStore.handleNewAnnotation({
+          newAnnotation,
+          newConnections: connections
+        });
+      }
+    );
+
+    // Display the new annotation
+    const newGeoJSAnnotation = this.createGeoJSAnnotation(newAnnotation);
+    this.annotationLayer.addAnnotation(newGeoJSAnnotation);
+  }
+
+  async addAnnotationFromSnapping(annotation: any) {
+    if (!annotation || this.maps.length !== 1) {
+      return;
+    }
+    const mapentry = this.maps[0];
+    const coordinates = annotation.coordinates();
+    this.annotationLayer.removeAnnotation(annotation);
+    const tool = this.selectedTool;
+    if (!tool) {
+      return;
+    }
+    const location = tool.values.annotation.coordinateAssignments;
+    if (!location) {
+      logError("Invalid snapping tool, annotation was not configured properly");
+      return;
+    }
+    const layer = location.layer;
```

Comment:
FYI, you can also have:
const {layer} = location;

Reply:
Thanks!